Source/WebCore/page/EventHandler.cpp

```diff
@@ -2794,17 +2794,17 @@ void EventHandler::updateMouseEventTargetNode(const AtomString& eventType, Node*
         page->imageOverlayController().elementUnderMouseDidChange(frame, m_elementUnderMouse.get());
 
     ASSERT_IMPLIES(m_elementUnderMouse, &m_elementUnderMouse->document() == frame->document());
+
+    if (m_lastElementUnderMouse && &m_lastElementUnderMouse->document() != frame->document()) {
```

Comment:
This seems like it’s happening too late. It should have happened right when the `m_lastElementUnderMouse` was moved to a new document, not waited until this function was called.

Reply:
Updated the code. PTAL. Thank you!